utils/decoder.py

```diff
@@ -0,0 +1,29 @@
+import serial
```

Comment:
I would suggest to put the python scripts in a separate PR, if you think they should be checked in at all.

Reply:
Having them checked in might be nice!  It would obviate the need for the `DEV_MODE_comms_handler`.

Alternately we handle everything in the GUI.